waiter/src/waiter/scheduler.clj

```diff
@@ -665,6 +666,32 @@
                         service
                         (assoc instances :active-instances active-instances))))))))
 
+(defn start-new-services-handler
+  [token-metric-chan-mult start-new-service-fn leader?-fn fallback-state-atom]
+  (cid/with-correlation-id
+    "start-new-services-goroutine"
+    (let [correlation-id (cid/get-correlation-id)
+          process-token-event-ch-buffer (async/sliding-buffer 1000)
+          process-token-event-ch
+          (async/chan
+            process-token-event-ch-buffer
+            (comp
+              (filter (fn leader? [_]
+                        (leader?-fn)))
```

Comment:
You can pass `leader?-fn` directly to filter. Though this feels strange. Maybe wrap in a when clause and guard execution?